Objective-C/Tests/ReplicatorTest.m

```diff
@@ -1773,6 +1773,91 @@ - (void) testPullFilter {
 }
 
 
+- (void) testReplicationEventExpirationWithPush {
+    NSError* error;
+    CBLMutableDocument* doc = [[CBLMutableDocument alloc] initWithID: @"doc"];
+    [doc setString: @"Tiger" forKey: @"species"];
+    [doc setString: @"Hobbes" forKey: @"pattern"];
+    Assert([self.db saveDocument: doc error: &error]);
+    
+    XCTestExpectation* expectation = [self expectationWithDescription: @"Document expiry test"];
+    id token1 = [self.db addDocumentChangeListenerWithID: doc.id
+                                                listener: ^(CBLDocumentChange *change)
+                 {
+                     AssertEqualObjects(change.documentID, doc.id);
+                     if ([change.database documentWithID: change.documentID] == nil) {
+                         [expectation fulfill];
+                     }
+                 }];
+    
+    // Push:
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    id config = [self configWithTarget: target type: kCBLReplicatorTypePush continuous: NO];
+    
+    __block id<CBLListenerToken> token;
+    __block CBLReplicator* replicator;
+    [self run: config reset: NO errorCode: 0 errorDomain: nil onReplicatorReady: ^(CBLReplicator* r) {
+        replicator = r;
+        token = [r addDocumentReplicationListener: ^(CBLDocumentReplication *docReplication) {
+            NSError* err;
+            Assert([self.db setDocumentExpirationWithID: doc.id
+                                             expiration: [NSDate date]
+                                                  error: &err]);
+        }];
+    }];
+    
+    [self waitForExpectationsWithTimeout: 5.0 handler: nil];
+    
+    AssertEqual(self.db.count, 0u);
+    AssertEqual(otherDB.count, 1u);
+    [self.db removeChangeListenerWithToken: token1];
+    [replicator removeChangeListenerWithToken: token];
+}
+
+
+- (void) testReplicationEventExpirationWithPull {
```

Comment:
I don't see the value of this test. Is this the same thing that .NET is testing?

Reply:
Test 5th is not valid for CBL. We don't have purge and become tombstone. What .NET has is test 4th which is testReplicationEventExpirationWithPush.

Can you rename the test `testReplicationEventExpirationWithPush` to `testPushAndForget`?

Reply:
I suggested removing testReplicationEventExpirationWithPull.

Reply:
Yes, will correct that.